community/logging/src/test/java/org/neo4j/logging/FormattedLogTest.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.logging;
+
+import org.junit.Test;
+
+import java.io.PrintWriter;
+import java.util.IllegalFormatException;
+
+import org.neo4j.function.Supplier;
+import org.neo4j.logging.FormattedLog.FormattedLogger;
+
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+
+public class FormattedLogTest
+{
+    @Test
+    public void loggerWriteLogShouldNotWriteAnythingForNullMessage()
+    {
+        // Given
+        PrintWriter writer = mock( PrintWriter.class );
+        FormattedLogger logger = newFormattedLogger();
+
+        try
+        {
+            // When
+            logger.writeLog( writer, null );
+            fail( "Should have thrown " + NullPointerException.class );
+        }
+        catch ( NullPointerException npe )
+        {
+            // Then
+            verifyZeroInteractions( writer );
+        }
+    }
+
+    @Test
+    public void loggerWriteLogShouldNotWriteAnythingForNullMessageAndNotNullThrowable()
+    {
+        // Given
+        PrintWriter writer = mock( PrintWriter.class );
+        FormattedLogger logger = newFormattedLogger();
+
+        try
+        {
+            // When
+            logger.writeLog( writer, null, new RuntimeException() );
+            fail( "Should have thrown " + NullPointerException.class );
+        }
+        catch ( NullPointerException npe )
+        {
+            // Then
+            verifyZeroInteractions( writer );
+        }
+    }
+
+    @Test
+    public void loggerWriteLogShouldWorkForNotNullMessageAndNullThrowable()
+    {
+        // Given
+        String message = "I need your clothes, your boots and your motorcycle";
+        Throwable throwable = null;
+
+        PrintWriter writer = mock( PrintWriter.class );
```

Comment:
Should this be a mock, or an actual class that outputs to a stringbuffer (on which we can assert the log line).